pootle/apps/pootle_store/migrations/0010_set_store_filetypes.py

```diff
@@ -0,0 +1,41 @@
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
+
+import logging
+
+from django.db import migrations, models
+
+from pootle.core.delegate import formats
+from pootle_format.exceptions import UnrecognizedFiletype
+from pootle_format.utils import ProjectFiletypes
+
+
+logger = logging.getLogger(__name__)
+
+
+def migrate_store_filetypes(apps, schema_editor):
+    format_registry = formats.get()
+
+    projects = apps.get_model("pootle_project.Project").objects.all()
+    stores = apps.get_model("pootle_store.Store").objects.all()
+    filetypes = apps.get_model("pootle_format.Format").objects.all()
+
+    for project in projects:
+        for filetype in project.filetypes.all():
+            stores = (
+                stores.filter(translation_project__project=project)
+                      .filter(filetype__isnull=True)
+                      .filter(name__endswith=".%s" % filetype.extension))
+            stores.update(filetype=filetype)
```

Comment:
not added for now - as there is some clarity required around handling Stores with no filetype (or for example changing the filetype of a Project). This will be addressed one way or another as the logic for this lands